## Задача

Написать код, который сможет:

Получить текст последних 20 постов со стены какого-то сообщества вк

у каждого поста есть id, записывайте его куда-нибудь

сохранить текст каждого поста в текстовый документ в папке, название которой совпадает с именем сообщества

После этого получить текст комментов к каждому посту
сохранить текст каждого коммента по пути id_поста/id_коммента.txt

Сложить всё это в в архив и отправить вам в телеграме.

## tg-бот

In [ ]:
!pip install -q telegram-send

     |████████████████████████████████| 409kB 7.6MB/s 
     |████████████████████████████████| 2.6MB 18.2MB/s 
     |████████████████████████████████| 61kB 6.9MB/s 


In [ ]:
!python /usr/local/lib/python3.6/dist-packages/telegram_send.py --configure-channel

Talk with the BotFather on Telegram (https://telegram.me/BotFather), create a bot and insert the token
❯ 1324984818:AAFobQf11jYEWjqFgSjqKs2047ne5jL2BkI
Connected with my_colab_progress_notifier_bot.

Do you want to send to a public or a private channel? [pub/priv]
❯ priv

Open https://web.telegram.org in your browser, sign in and open your private channel.
Now copy the URL in the address bar and enter it here:
❯ https://web.telegram.org/#/im?p=c1284475359_8938296753326905798

Congratulations! telegram-send can now post to your channel!


Talk with the BotFather on Telegram (https://telegram.me/BotFather), create a bot and insert the token

❯ 1324984818:AAFobQf11jYEWjqFgSjqKs2047ne5jL2BkI

Connected with my_colab_progress_notifier_bot.

Do you want to send to a public or a private channel? 

[pub/priv]

❯ priv

Open https://web.telegram.org in your browser, sign in and open your private channel.
Now copy the URL in the address bar and enter it here:

❯ https://web.telegram.org/#/im?p=c1284475359_8938296753326905798

Congratulations! telegram-send can now post to your channel!

In [ ]:
!python /usr/local/lib/python3.6/dist-packages/telegram_send.py 'Привет!'

## Запросы к vk api

### Пусть будет группа https://vk.com/bowie_salad
#### Нет, там в основном картинки без текста, но группа замечательная

### Тогда будет родной звук: https://vk.com/rodzvuk

In [ ]:
import requests

In [ ]:
import os

In [ ]:
VK_API_TOKEN = "" # сюда нужно вставить токен авторизованного приложения

### найдем id по названию группы

In [ ]:
resolved_id_by_name = requests.get(
    'https://api.vk.com/method/utils.resolveScreenName',
     params={
        'screen_name': 'rodzvuk',
        'v': '5.92',
        'access_token': VK_API_TOKEN
     }).json()["response"]["object_id"]

In [ ]:
resolved_id_by_name

83415396

### метод wall.get для получения 20 постов из группы

In [ ]:
posts = requests.get('https://api.vk.com/method/wall.get',
                    params={
                        "owner_id": -resolved_id_by_name,
                        "v":"5.94",
                        "access_token": VK_API_TOKEN,
                        "count": 20
                    }).json()['response']['items']

### Получим тексты постов и сохраним их в папке rodzvuk

In [ ]:
!mkdir rodzvuk

In [ ]:
!ls

sample_data


In [ ]:
for post in posts:
  with open(f"rodzvuk/{post['id']}.txt", "w") as f:
    f.write(post["text"])
  os.mkdir(f"rodzvuk/{post['id']}")
  # получаем комменты по id поста:
  comments = requests.get('https://api.vk.com/method/wall.getComments',
                      params={
                          "owner_id": -resolved_id_by_name,
                          "v":"5.94",
                          "post_id": post['id'],
                          'access_token': VK_API_TOKEN
                      }).json()["response"]["items"]
  
  for comment in comments:
    with open(f"rodzvuk/{post['id']}/{comment['id']}.txt", "w") as f:
      f.write(comment["text"])

##### загрузили тексты 20 постов

In [ ]:
!ls rodzvuk

280443	    280526.txt	280605	    280667.txt	280761	    280793.txt
280443.txt  280542	280605.txt  280719	280761.txt  280799
280460	    280542.txt	280622	    280719.txt	280784	    280799.txt
280460.txt  280571	280622.txt  280737	280784.txt  280807
280475	    280571.txt	280640	    280737.txt	280789	    280807.txt
280475.txt  280586	280640.txt  280752	280789.txt
280526	    280586.txt	280667	    280752.txt	280793


#### Добавили файлы с комментами

In [ ]:
!ls rodzvuk/280443

280450.txt  280457.txt	280464.txt  280474.txt	280518.txt  280711.txt


##### Рекурсивно заархивируем папку rodzvuk

In [ ]:
!zip -r rodzvuk.zip rodzvuk 

  adding: rodzvuk/ (stored 0%)
  adding: rodzvuk/280799/ (stored 0%)
  adding: rodzvuk/280799/280817.txt (deflated 27%)
  adding: rodzvuk/280799/280803.txt (stored 0%)
  adding: rodzvuk/280799/280800.txt (stored 0%)
  adding: rodzvuk/280799/280805.txt (stored 0%)
  adding: rodzvuk/280799/280804.txt (stored 0%)
  adding: rodzvuk/280799/280814.txt (stored 0%)
  adding: rodzvuk/280799/280801.txt (stored 0%)
  adding: rodzvuk/280799/280802.txt (stored 0%)
  adding: rodzvuk/280799/280806.txt (stored 0%)
  adding: rodzvuk/280799/280816.txt (deflated 12%)
  adding: rodzvuk/280667/ (stored 0%)
  adding: rodzvuk/280667/280677.txt (stored 0%)
  adding: rodzvuk/280667/280670.txt (stored 0%)
  adding: rodzvuk/280667/280679.txt (deflated 17%)
  adding: rodzvuk/280667/280693.txt (deflated 20%)
  adding: rodzvuk/280667/280708.txt (deflated 15%)
  adding: rodzvuk/280526/ (stored 0%)
  adding: rodzvuk/280526/280709.txt (stored 0%)
  adding: rodzvuk/280526/280527.txt (stored 0%)
  adding: rodzvuk/280526

##### Отправим результат через tg-bot

In [ ]:
!python /usr/local/lib/python3.6/dist-packages/telegram_send.py --file rodzvuk.zip --caption "Архив с постами"